supabase/functions/_script/backup_and_cleanup.ts

```diff
@@ -0,0 +1,103 @@
+import { createClient } from 'https://esm.sh/@supabase/supabase-js@^2.2.3';
+import 'https://deno.land/x/dotenv/load.ts';
+import { r2 } from '../_utils/r2.ts';
+import type { Database } from '../_utils/supabase.types.ts';
+import * as fs from 'https://deno.land/std/fs/mod.ts';
+import * as path from 'https://deno.land/std/path/mod.ts';
+
+const supabaseUrl = Deno.env.get('SUPABASE_URL');
+const supabaseServiceRoleKey = Deno.env.get('SUPABASE_SERVICE_ROLE_KEY');
+const backupFolder = 'backup';
+
+if (!supabaseUrl || !supabaseServiceRoleKey) {
+  console.error('Error: SUPABASE_URL and SUPABASE_SERVICE_ROLE_KEY environment variables are required.');
+  Deno.exit(1);
+}
+
+function useSupabase() {
+  const options = {
+    auth: {
+      autoRefreshToken: true,
+      persistSession: true,
+      detectSessionInUrl: false,
+    },
+  };
+  return createClient<Database>(supabaseUrl, supabaseServiceRoleKey, options);
+}
+
+async function backupAndDeleteOldEntries() {
+  const currentDate = new Date();
+  const backupFilenames: string[] = [];
+  const pageSize = 1000;
+
+  try {
+    if (!await fs.exists(backupFolder)) {
+      await fs.ensureDir(backupFolder);
+    }
+
+    const supabase = useSupabase();
+    const daysToKeep = 30;
+    let offset = 0;
+    let hasMoreData = true;
+    
+    const cutoffDate = new Date(currentDate.getTime() - daysToKeep * 24 * 60 * 60 * 1000); // Declare cutoffDate outside the loop
+
+    while (hasMoreData) {
+      const query = `
+        SELECT * FROM app_versions
```

Comment:
i think there issue here.
The table we talk about is stats and devices not app_versions

Reply:
I have updated the code Please have a look.